### PR TITLE
[language server] Allow configuration of validation rules.

### LINKED ISSUE #1056: Add support for custom validation rules in apollo-language-server
Inspired by #800 

> NOTE: I **won't** be personally building this anytime soon. I WILL, however, gladly review any pull requests that adds this feature 😄 

Right now, the `apollo-language-server` runs a set of validation rules against operations (great, right??). 

In certain circumstances (like if you're running a relay project), you may want to disable existing validations or add new ones (like requiring an `id` on every object type for caches).

Since the `apollo.config.js` is an executable file, not just a static configuration, we can actually add these validations to it to run against client projects!

Here's my proposed API:

```
// apollo.config.js
const { defaultValidationRules } = require('apollo/lib/defaultValidationRules')
const relayValidationRules = require('my-custom-validation-rules')

module.exports = {
  client: {
    validationRules: [
		relayValidationRules,
		...defaultValidationRules.filter(removeBitsThatMessUpRelay)
	]
  }
}
```

This way, you'd have _full_ control (for better or worse 😉) over validation rules, rather than just enabling/disabling rules from graphql-js

---

This would require changes to: 
- apollo-language-server
	- add validation overrides to the config/validation
- apollo
	- export the default validation rules for easy import

### PR DESCRIPTION
**Pull request by alloy**
Closes #1056

(I didn’t find any test coverage for the areas I touched, but am pretty sure my additions are backwards compatible.)

### Before

vscode-apollo before

### Config

vscode-apollo config

### After

vscode-apollo after

### CODE REVIEW

**Comment by alloy** [comment]
I've seen that CI failure before and it went away after a re-run by fixing a spelling error in a comment and force pushing. I can do the same again, but perhaps it’s better to highlight this gremlin?

**Comment by alloy** [comment]
Oops, forgot to mention you before, @JakeDawkins.

File: package-lock.json [hunk 1]
```diff
@@ -2327,17 +2327,15 @@
       "dependencies": {
         "@oclif/config": {
           "version": "1.13.0",
-          "resolved": "https://registry.npmjs.org/@oclif/config/-/config-1.13.0.tgz",
```
Comment (janicduplessis) [inline package-lock.json]: Should probably revert the changes here, I guess its caused by using a diff npm version

Comment (alloy) [reply package-lock.json]: Good call, thanks!

**Comment by alloy** [comment]
I added the ability to specify a function that can be used to filter the default validation rules. The reason for this is that when using the extension on non-javascript projects, there is no good way to import the default validation rules from the Apollo package without adding a `package.json` file, the `apollo` dependency, and installing the node modules. This feels like overkill for such projects.

In our case, we have Ruby applications that we want to use the extension on, but we want to filter out the `NoAnonymousQueries` validation.

**Comment by trevor-scheer** [comment]
This LGTM. There are a couple tasks I'd like to spin out of this.
* Export the rule set at the top level of the language server (and maybe `apollo`)
* Document the new config option (see referenced issue on this PR)

If you're up for both or either, just let me know - otherwise I'll take care of them. Thanks again!

**Comment by trevor-scheer** [comment]
@alloy FYI, I cut a release but can't get the extension working (unrelated to your changes) so I'll need to get a fix together before I can get that packaged up and posted on the marketplace. Sorry about the delay! I'm aiming for tomorrow, but I'll keep you looped in either way.

**Comment by trevor-scheer** [comment]
For context on the issue, see: https://github.com/apollographql/apollo-tooling/pull/1306
Extension is now released and up on the marketplace `vscode-apollo@1.7.1` 🎉

Thanks again!

**Comment by alloy** [comment]
Awesome, thanks!! 🙏

Yeah I can do those other tasks 👍
